airbyte-integrations/connectors/source-postgres/integration_tests/seed/hook.py

```diff
@@ -0,0 +1,212 @@
+        connection.rollback()
+    finally:
+        cursor.close()
+
+def generate_schema_date_with_suffix():
+    current_date = datetime.datetime.now().strftime("%Y%m%d")
+    suffix = ''.join(random.choices(string.ascii_letters + string.digits, k=8))
+    return f"{current_date}-{suffix}"
+
+def prepare():
+    schema_name = generate_schema_date_with_suffix()
+    with open("./generated_schema.txt", "w") as f:
+        f.write(schema_name)
+
+def setup():
+    schema_name = load_schema_name_from_catalog()
+    write_supporting_file(schema_name)
+    table_name = "id_and_name_hook"
+
+    # Define the records to be inserted
+    records = [
+        ('1', 'one'),
+        ('2', 'two'),
+        ('3', 'three')
+    ]
+
+    # Connect to the database
+    connection = connect_to_db()
+
+    if connection:
+        # Create the schema
+        create_schema(connection, schema_name)
+        create_table(connection, schema_name, table_name)
+
+        # Insert the records
+        insert_records(connection, schema_name, table_name, records)
+
+        # Close the connection
+        connection.close()
+
+def load_schema_name_from_catalog():
+    with open("./generated_schema.txt", "r") as f:
+        return f.read()
+
+def remove_all_write_files():
+    print("***removing write files***")
+    os.remove(catalog_write_file)
+    os.remove(catalog_view_write_file)
+    os.remove(catalog_incremental_write_file)
+    os.remove(abnormal_state_write_file)
+
+def delete_schemas_with_prefix(conn, date_prefix):
+    try:
+        # Connect to the PostgreSQL database
+        conn.autocommit = True
+        cursor = conn.cursor()
+
+        # Query to find all schemas that start with the specified date prefix
+        query = sql.SQL("""
+            SELECT schema_name
+            FROM information_schema.schemata
+            WHERE schema_name LIKE %s;
+        """)
+
+        cursor.execute(query, (f"{date_prefix}%",))
+        schemas = cursor.fetchall()
+
+        # Generate and execute DROP SCHEMA statements for each matching schema
+        for schema in schemas:
+            drop_query = sql.SQL("DROP SCHEMA IF EXISTS {} CASCADE;").format(sql.Identifier(schema[0]))
+            cursor.execute(drop_query)
+            print(f"Schema {schema[0]} has been dropped.")
+
+    except Exception as e:
+        print(f"An error occurred: {e}")
+    finally:
+        if cursor:
+            cursor.close()
+        if conn:
+            conn.close()
+
+def teardown():
+    connection = connect_to_db()
+    today = datetime.datetime.now()
+    yesterday = today - timedelta(days=1)
+    formatted_yesterday = yesterday.strftime('%y%m%d')
+    delete_schemas_with_prefix(connection, formatted_yesterday)
+    remove_all_write_files()
+
+if __name__ == "__main__":
+    command = sys.argv[1]
+    if command == "setup":
+        setup()
+    elif command == "teardown":
+        teardown()
+    elif command == "prepare":
+        prepare()
```

Comment:
to simplify, maybe we shall just make prepare() as part of setup()? setup to me means all the preparation work...

Reply:
prepare is supposed to run once before we start CAT test, while setup is going to run before each test.

Reason being we realized it's easier to work on a single schema among the entire test suite, to avoid that async incremental test conflictions.